browser_history/cli.py

```diff
@@ -75,6 +74,27 @@ def make_parser():
                 If not provided, standard output is used.""",
     )
 
+    parser_.add_argument(
+        "-p",
+        "--profile",
+        default=None,
+        help="""
+                Specify the profile from which to fetch history or bookmarks. If
+                not provided all profiles are fetched
+        """,
+    )
+
+    parser_.add_argument(
+        "--show-profiles",
+        default=None,
+        metavar="BROWSER",
+        help=f"""
+                List all available profiles for a given browser where browser
+                can be on of all, default, {AVAILABLE_BROWSERS}. The browser
+                must always be provided.
+        """,
+    )
```

Comment:
IMO we should use the `-b`/`--browser` to get the browser name and use `--show-profiles` just as a boolean flag - whether to show profiles or not.

Other than the semantics, it adds too much clutter in the help page.

Having the `-b` argument when this flag is used can be ensured by using something like [this](https://stackoverflow.com/a/19414853/11199009).

Reply:
This will also avoid the issue when an invalid browser is passed to `--show-profiles` it shows an entire stacktrace.
```powershell
(base) PS D:\DEV\Others\OpenSource\browser-history> browser-history --show-profiles brr
ERROR: brr browser is unavailable. Check --help for available browsers
Traceback (most recent call last):
  File "D:\APPS\Miniconda\Scripts\browser-history-script.py", line 33, in <module>
    sys.exit(load_entry_point('browser-history', 'console_scripts', 'browser-history')())
  File "d:\dev\others\opensource\browser-history\browser_history\cli.py", line 198, in main
    cli(sys.argv[1:])
  File "d:\dev\others\opensource\browser-history\browser_history\cli.py", line 116, in cli
    if not browser_class.profile_support:
AttributeError: 'NoneType' object has no attribute 'profile_support'
```

Reply:
I thought of doing it that way but felt that allowing `--show-profiles` to work more than just a flag was easier. This allows the user not to have to provide two options. It's also the reason I moved browser fetching to it's own function. It's also more intuitive honestly.

```bash
browser-history --show-profiles edge
```
```bash
browser-history -b edge --show-profiles
```

It's easier to fix the full stack-trace issue than rewrite the CLI code like you want, took me several hours to wire up the CLI code as it is (you know how hairy the CLI code gets and it's easy to miss something). Adding the code below to line 116 should fix it 

```python
if browser_class is None:
      sys.exit(1)
```

Reply:
Also, this allows the help page to specify that "all" cannot be used as an option for `BROWSER` when it comes to --show-profiles as opposed to when the browser parameter is supplied through `-b/--browser` option which (according to usage allows "all" as a valid parameter)

Reply:
Very good points, you have convinced me xD.

It does look more intuitive and the fact that "all" cannot be used outweigh the benefits of the approach I suggested. Thank you for the effort 😄

Reply:
Good communication skills I guess 😁😁